gitlab-runner/src/client.rs

```diff
@@ -106,6 +106,7 @@ struct JobUpdate<'a> {
 
 #[derive(Debug, Clone)]
 pub struct JobUpdateReply {
+    #[allow(dead_code)]
```

Comment:
I'd put a comment here, something like:

```suggestion
    // on every request Gitlab is tell you what kind of update rate it expects from runner -> server for a given job
    // we currently only act on the indications we get back from doing trace calls (e.g. appending to the log)
    // but GitLab also sends it when updating the job metadata so keep the unused field around to at least document the fact
    #[allow(dead_code)]
```